src/_pytest/runner.py

```diff
@@ -220,20 +235,29 @@ def result(self):
         return self._result
 
     @classmethod
-    def from_call(cls, func, when, reraise=None) -> "CallInfo":
+    def from_call(cls, func, when, item=None, reraise=None) -> "CallInfo":
         #: context of invocation: one of "setup", "call",
         #: "teardown", "memocollect"
-        start = time()
-        excinfo = None
+        call = cls(
```

Comment:
this structure changes a `CallInfo` from something concrete and state-free to a state-machine tracking the progress if a call

i believe a class name change is warranted to make this clear, but off hand no good name describing it comes to mind